src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -27,13 +27,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<String> userEmail = new ArrayList<String>();
```

Comment:
What is the reason of having a list for userEmail? Why not just a single String?

Any anytime, there is only one user logged in, it is got from userService.getCurrentUser().getEmail(), and is what we would like to display on the website and check authorization against with. 

This likely caused the problem cause before user logged in, "Stranger" is always added into the list, and as the first one, in your javascript, you are only checking the first element in the list. 

I would also recommend some System.out.println in your local java code and check the user email string. Please remove the println before you commit.

Reply:
Initially I planned to include more information about the user but since the email is the only thing that is currently being stored I have changed this into a String. Instead of adding to a list the string now changes to either "Stranger" if the user is not logged in, the user's email, or "deny" if the user does not have access.